rmgpy/tools/thermo_group_validation.py

```diff
@@ -0,0 +1,237 @@
+    heading = ''
+    units =''
+    if variable == 'H':
+        heading = 'Enthalpy'
+        units = 'kcal/mol'
+    elif variable == 'S':
+        heading = 'Entropy'
+        units = 'cal/mol'
+    elif variable == 'Cp':
+        heading = 'Heat_Capacity'
+        units = 'cal/mol'
+    elif variable == 'G':
+        heading = 'Free_Energy'
+        units = 'kcal/mol'
+    T0 = inputDict['temperature']
+    #load database with no libraries
+
+    #change to new directory for RMG-database
+    with cd(settings['database.directory']):
+        for commitStr in inputDict['commitStrs']:
+            with previousCommit(commitStr, settings['database.directory']):
+                newGroups= ThermoDatabase()
+                newGroups.load(os.path.join(settings['database.directory'], 'thermo'))
+                for smiles in speciesThermo:
+                    newSpecies = Species().fromSMILES(smiles)
+                    speciesThermo[smiles].append(newGroups.getThermoDataFromGroups(newSpecies))
+
+    parityData={}
+
+    #get initial dictionary value: [data from library, data from groups of initial commit]
+    for smiles, thermoList in speciesThermo.iteritems():
+        parityData[smiles] = []
+        for thermoData in thermoList:
+            if isinstance(thermoData, LibraryEntry):
+                parityData[smiles].append(parityFunction(variable, thermoData.thermoData, T0))
+            else: parityData[smiles].append(parityFunction(variable, thermoData, T0))
+
+    #make list of highst stdDev of groups (first element is name, second is library value)
+    csvList = [[name]+data for name, data in parityData.iteritems()]
+
+    #Put into kcal/cal
+    conversionCsvList=[]
+    for point in csvList:
+        newPoint=[point[0]]
+        for value in point[1:]:
+            if variable == 'H' or variable == 'G':
+                newPoint.append(value/1000.0/4.182)
+            else:
+                newPoint.append(value/4.182)
+        conversionCsvList.append(newPoint)
+
+    csvList= conversionCsvList
+
+    #Sort based on largest difference in group estimation
+    csvList = sorted(csvList, key = lambda x: std(x[2:]))
+    csvList.reverse()
+
+    with open(os.path.join(os.getcwd(), heading+'.csv'), 'wb') as output:
+        spamwriter=csv.writer(output)
+        #add Header
+        spamwriter.writerow(['SMILES', 'Library Value at '+str(T0)+'K']+ inputDict['commitNames'])
+        for line in csvList:
+            spamwriter.writerow(line)
+
+    #Remove points where group estimates are the same for plotting
+    prunedCsvList=[]
+    #don't include header in pruned csv
+    for item in csvList:
+        if not std(item[2:]) == 0:
+            prunedCsvList.append(item)
+
+    xvar = [x+1 for x in range(len(prunedCsvList))]
+    for index, x in enumerate(range(len(prunedCsvList[0])-2)):
+        yvar = [abs(x[1]-x[index+2]) for x in prunedCsvList]
+        plt.semilogy(xvar, yvar, 's', label = inputDict['commitNames'][index])
```

Comment:
Yeah it probably depends on what the user wants. I was thinking the main purpose is to compare between commits, so the prunedCsvList ignores entries where the group additivity has not changed. 

It could be argued that we just want to generally visualize how good our group additivity works. In that case, we would not want to use the prunedCsvList. Earlier I found that graphs get pretty busy when putting in multiple libraries though.